NomadUI/Widgets/NUIButton.cpp

```
@@ -54,7 +69,21 @@ void NUIButton::onRender(NUIRenderer& renderer) {
-    renderer.fillRoundedRect(bounds, radius, bgColor);
+    // FLAT DESIGN: Only draw background if hovered, pressed, or explicitly set
+    // This removes the "light square boxes" for a cleaner look
+    if (isHovered() || pressed_ || useCustomColors_) {
+        NUIColor drawColor = bgColor;
+        
+        // Refined hover style: Use subtle transparency instead of solid block
+        // This gives the "Ableton-like" feel without the "boxy" look
+        if (!useCustomColors_ && isHovered() && !pressed_) {
+            // Use a very low opacity for the hover state
+            // This assumes the theme hover color is meant to be an overlay
+            drawColor = drawColor.withAlpha(0.15f); 
```

Clang (14.0.6)

[warning] 0.15f is a magic number; consider replacing it with a named constant

(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)